packages/fuel-gauge/src/abi/abi-coder.test.ts

```diff
@@ -0,0 +1,2634 @@
+import { bn, FuelError, getRandomB256 } from 'fuels';
+import type { AssetId, BigNumberish, EvmAddress, RawSlice, WalletUnlocked } from 'fuels';
+import { expectToThrowFuelError, launchTestNode } from 'fuels/test-utils';
+
+import { AbiContractFactory } from '../../test/typegen';
+import type { AbiContract } from '../../test/typegen';
+import {
+  EnumWithNativeInput,
+  EnumWithNativeOutput,
+  ExternalEnumInput,
+} from '../../test/typegen/contracts/AbiContract';
+import type {
+  EnumWithBuiltinTypeInput,
+  EnumWithBuiltinTypeOutput,
+  EnumWithVectorInput,
+  EnumWithVectorOutput,
+  IdentityInput,
+  IdentityOutput,
+  StructDoubleGenericInput,
+  StructSimpleInput,
+  StructSimpleOutput,
+  StructWithGenericArrayInput,
+  StructWithMultiOptionInput,
+  StructWithMultiOptionOutput,
+  StructCInput,
+  StructWithNestedArrayInput,
+  StructWithNestedTupleInput,
+  StructSingleGenericInput,
+  StructWithImplicitGenericsInput,
+  AssetIdInput,
+  StructWithEnumArrayInput,
+  StructWithEnumArrayOutput,
+  StructWithSingleOptionOutput,
+  StructWithSingleOptionInput,
+} from '../../test/typegen/contracts/AbiContract';
+import type { Option, Result, Vec } from '../../test/typegen/contracts/common';
+
+import {
+  U16_MAX,
+  U16_MIN,
+  U256_MAX,
+  U256_MIN,
+  U32_MAX,
+  U32_MIN,
+  U64_MAX,
+  U64_MIN,
+  U8_MAX,
+  U8_MIN,
+} from './constants';
+import { toEqualBn } from './vitest.matcher';
+
+expect.extend({ toEqualBn });
+
+/**
+ * @group node
```

Comment:
Could and should this be run in browser?

Reply:
These are standard contract integration tests so it must be runnable in the browser.

Reply:
f1397348bf37e983380d4d59fa0203e5fc3404ad